contrib/DEBUGGING.md

```diff
@@ -0,0 +1,133 @@
+Sure, here's a more detailed guide on debugging Bittensor, including some code examples from the Bittensor repository.
+
+## Installation
+
+First, make sure you have Bittensor installed correctly. There are three ways to install Bittensor:
+
+1. Through the installer:
+
+```bash
+/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/opentensor/bittensor/master/scripts/install.sh)"
+```
+
+2. With pip:
+
+```bash
+pip install bittensor
+```
+
+3. From source:
+
+```bash
+git clone https://github.com/opentensor/bittensor.git
+python3 -m pip install -e bittensor/
+```
+
+You can test your installation by running:
+
+```bash
+python3 -c "import bittensor; print(bittensor.__version__)"
+```
+
+## Wallets
+
+Bittensor uses wallets for identity and ownership. Wallets consist of a coldkey and hotkey. Coldkeys store funds securely and operate functions such as transfers and staking, while hotkeys are used for all online operations such as signing queries, running miners, and validating.
+
+Here's how to create a wallet using the Python API:
+
+```python
+import bittensor as bt
+wallet = bt.wallet()
+wallet.create_new_coldkey()
+wallet.create_new_hotkey()
+print(wallet)
+```
+
+## Querying the Network
+
+You can query the Bittensor network using the Python API. Here's an example of how to do this:
+
+```python
+import bittensor as bt
+
+# Query through the foundation endpoint.
+print(bt.prompt("Heraclitus was a "))
+```
+
+## Debugging Miners
+
+Miners in Bittensor are incentivized to contribute distinct forms of value determined by the verification mechanism that that subnetwork’s Validators are running. 
+
+Here's an example of how to register a miner:
+
+```bash
+btcli register --netuid <subnetwork uid>
+```
+
+Once registered, the miner attains a slot specified by their UID. To view your slot after registration, run the overview command:
+
+```bash
+btcli overview --netuid <subnetwork uid>
+```
+
+Registered miners can select from a variety of pre-written miners or write their own using the Python API. Here's an example of how to run a miner:
+
+```bash
+python3 bittensor/neurons/text_prompting/miners/GPT4ALL/neuron.py --netuid 1
+```
+
+## Debugging Validators
+
+Validators in Bittensor are participants who hold TAO. They use a dual proof-of-stake, proof-of-work mechanism called Yuma Consensus. Here's how to stake funds:
+
+```bash
+btcli stake --help
+```
+
+And here's how to become a delegate available for delegated stake:
+
+```bash
+btcli nominate --help
+```
+
+## Using the CLI
+
+Bittensor comes with a command-line interface (CLI) called `btcli` that you can use to interact with the network. Here's how to get help on the available commands:
+
+```bash
+btcli --help
+```
```

Comment:
what's the relation of this with _**debugging**_?

Reply:
Oops!  These were placeholders I forgot to update.

Updated the debug file to be releavnt: eb0cd04f678fdc75e05ffc77d0c6d0c01bdb7111